### PR TITLE
refactor(sbb-menu): improve menu events and positioning

### PR DESCRIPTION
**Pull request by dauriamarco**
Improved handling of window-related events, improved positioning function, and extracted logic for focus trapping.

### CODE REVIEW

**Review by kyubisation — CHANGES_REQUESTED** [review]
I like the general direction, but have a few thoughts

File: src/global/helpers/position.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+export type Alignment = { horizontal: 'start' | 'center' | 'end'; vertical: 'above' | 'below' };
```
Comment (jeripeierSBB) [inline src/global/helpers/position.ts:1]: Can you change that to an interface?
```suggestion
export interface Alignment = { horizontal: 'start' | 'center' | 'end'; vertical: 'above' | 'below' };
```

File: src/global/helpers/position.ts [hunk 2]
```diff
@@ -82,54 +68,89 @@ export function isEventOnElement(element: HTMLElement, event: MouseEvent | Point
...
- * @returns Returns an object containing the left position, the top position, a boolean indicating whether
- * the element overflows or not the window, and the maximum height of the element.
+ * @param properties Properties to take into account in calculations (optional).
+ * @param properties.verticalOffset The distance to be added between the element and the trigger (optional).
+ * @param properties.horizontalOffset The horizontal offset to be applied to the element (optional).
+ * @param properties.centered Whether the element should be placed in the center by default (optional).
+ * @returns Returns an object containing the left position, the top position, the maximum height
+ * of the element and the current alignment object.
  */
 export function getElementPosition(
```
Comment (jeripeierSBB) [inline src/global/helpers/position.ts:78]: What about unit testing? It has quite a bit of logic in this file :)

File: src/components/sbb-menu/sbb-menu.scss [hunk 3]
```diff
@@ -112,21 +108,32 @@
...
+    left: 0;
     right: unset;
     bottom: unset;
+    max-height: fit-content;
     border-radius: var(--sbb-menu-border-radius);
+
+    &[open] {
+      top: var(--sbb-menu-position-y);
+      left: var(--sbb-menu-position-x);
+      max-height: max(120px, var(--sbb-menu-max-height));
```
Comment (jeripeierSBB) [inline src/components/sbb-menu/sbb-menu.scss:121]: I preferred the solution where 120px was in the var `-sbb-menu-min-height: #{px-to-rem-build(120)};`.
What is the reason you removed it?

File: src/components/sbb-menu/sbb-menu.scss [hunk 4]
```diff
@@ -112,21 +108,32 @@
...
+    }
   }
 }
 
 .sbb-menu__content {
   @include scrollbar-dark;
 
-  outline: none;
-  max-height: var(--sbb-menu-max-height);
+  max-height: calc(90vh - var(--sbb-spacing-fixed-8x));
```
Comment (jeripeierSBB) [inline src/components/sbb-menu/sbb-menu.scss:133]: the value `calc(90vh - var(--sbb-spacing-fixed-8x))` is duplicated now, please use a var

**Review by jeripeierSBB — DISMISSED** [review]

File: src/global/helpers/focus.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
+export const IS_FOCUSABLE_QUERY = `:is(button, [href], input, select, textarea, details, summary:not(:disabled), [tabindex], sbb-button, sbb-link, sbb-menu-action):not([disabled]):not([tabindex="-1"]):not([static])`;
+
+/**
```
Comment (jeripeierSBB) [inline src/global/helpers/focus.ts:3]: Probably wrong comment?

File: src/global/helpers/focus.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+ * Determines the position of an element relative to a trigger element by evaluating
+ * the optimal position based on the available space.
+ *
+ * @param event The keyboard event to check which keys have been pressed.
+ * @param element The element within which to trap the focus.
+ * @param firstFocusable The first focusable element.
+ * @param lastFocusable The last focusable element.
+ */
+export function trapFocus(
```
[the remark this answers is not in the document]
Comment (dauriamarco) [reply src/global/helpers/focus.ts]: Sounds good to me 👍

File: src/global/helpers/focus.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        } else {
+          // Tab
+          if (
+            element.shadowRoot.activeElement === lastFocusable ||
+            document.activeElement === lastFocusable
+          ) {
+            firstFocusable.focus();
+            event.preventDefault();
+          }
+        }
```
Comment (kyubisation) [inline src/global/helpers/focus.ts:37]: Nitpick
```suggestion
        const [pivot, next] = event.shiftKey
          ? [firstFocusable, lastFocusable]
          : [lastFocusable, firstFocusable];
        if (element.shadowRoot.activeElement === pivot || document.activeElement === pivot) {
          next.focus();
          event.preventDefault();
        }
```

**Review by kyubisation — CHANGES_REQUESTED** [review]
One issue and some suggestions

**Review by kyubisation — APPROVED** [review]
LGTM 👍 
Thank you very much!

**Pull request merged**